source/scripts/app.js

```diff
@@ -51,6 +105,54 @@ function createExplorePage() {
     "recipe-cards-section"
   );
 
+  //Get references to search bar on explore
+  let shadow = explorePage.shadowRoot;
+  let input = shadow.getElementById("search-bar");
+  let vegan = shadow.getElementById("vegan");
+  let glutenFree = shadow.getElementById("gluten-free");
+  let vegetarian = shadow.getElementById("vegetarian");
+  input.addEventListener("keyup", (e) => {
+    if (e.key === "Enter") {
+      e.preventDefault();
+      if (
+        input.value !== "" ||
+        vegan.checked ||
+        glutenFree.checked ||
+        vegetarian.checked
+      ) {
+        if (
+          shadow
+            .getElementById("explore-top-level")
+            .classList.contains("type-explore")
+        ) {
+          toggleExplorePageType();
+        }
+        let queryObj = {};
+        queryObj.query = input.value;
+        queryObj.diet = "";
+        if (vegan.checked) {
+          queryObj.diet += "vegan ";
+        }
+        if (glutenFree.checked) {
+          queryObj.diet += "gluten free ";
+        }
+        if (vegetarian.checked) {
+          queryObj.diet += "vegetarian ";
+        }
+        populateExplorePage(queryObj);
+      } else {
+        if (
+          !shadow
+            .getElementById("explore-top-level")
+            .classList.contains("type-explore")
+        ) {
+          toggleExplorePageType();
+        }
+        populateExplorePage();
+      }
+    }
+  });
+
```

Comment:
Do you think we should move this logic out from `createExplorePage()` and into another function like `bindExploreSearchBar()`?

Reply:
Sure, I think that would work. I was also thinking about changing the .checked stuff to be a for loop, but I think I got null when querySelectorAll'ing the classes. Both ideas could help for neater code.